files/build_templates/docker_image_ctl.j2

```diff
@@ -659,6 +659,24 @@ start() {
     else
         SMARTSWITCH_MNT=""
     fi
+    # Dynamic sensors.conf setup: Create symlink from hw-management to SONiC expected location
+    # This allows pmon container to use the latest sensor configuration from hw-management
+    HW_MGMT_SENSORS_CONF="/var/run/hw-management/config/lm_sensors_config"
+    SONIC_SENSORS_CONF_LINK="/usr/share/sonic/device/$PLATFORM/sensors.conf"
+    SONIC_SPECIFIC_SENSORS_CONF="/usr/share/sonic/device/$PLATFORM/sonic_specific_sensors.conf"
+    if [ -e "$HW_MGMT_SENSORS_CONF" ] && [ ! -e "$SONIC_SENSORS_CONF_LINK" ]; then
+        sudo ln -s "$HW_MGMT_SENSORS_CONF" "$SONIC_SENSORS_CONF_LINK"
+        if [ ! -L "$SONIC_SENSORS_CONF_LINK" ]; then
+            echo "Error: Failed to create symlink: $SONIC_SENSORS_CONF_LINK -> $HW_MGMT_SENSORS_CONF, 'sensors' command will print raw sensors data"
+            break
+        fi
+        if [ -e "$SONIC_SPECIFIC_SENSORS_CONF" ]; then
+            if ! sudo sh -c "cat '$SONIC_SPECIFIC_SENSORS_CONF' >> '$SONIC_SENSORS_CONF_LINK'"; then
+                echo "Error: Failed to append SONiC-specific sensors.conf"
+                return 1
```

Comment:
please reconsider the return 1 - you may keep the sensors file without platform specific code